dataflux_pytorch/dataflux_mapstyle_dataset.py

```diff
@@ -200,7 +200,8 @@ def _list_GCS_blobs_with_retry(self):
                     prefix=self.config.prefix,
                     retry_config=self.config.list_retry_config,
                 )
-                lister.client = self._get_or_create_storage_client()
+                if self.storage_client is not None:
```

Comment:
```suggestion
                # If the dataset was not initialized with an storage_client, ensure that we do not attach a client to the lister to avoid pickling errors (#58).
                if self.storage_client is not None:
```

Reply:
Done.